### PR TITLE
QGB remove dependency on geth dependency for encoding the data root

### PR DESCRIPTION
**Pull request by amityadav0**
## Description

- Remove dependency on geth dependency for encoding the data root
- Manually pad the hex representation of the height to 32 bytes and concatenate the data root to it.

Closes: #934

### CODE REVIEW

**Review by SweeXordious — CHANGES_REQUESTED** [review]

File: rpc/core/blocks.go [hunk 1]
```diff
@@ -177,59 +177,31 @@ func DataRootInclusionProof(
...
+	if l == length {
+		return byt
+	}
+	tmp := make([]byte, length)
+	copy(tmp[length-l:], byt)
+	return tmp
+}
+
 // EncodeDataRootTuple takes a height and a data root and returns the equivalent of
-// `abi.encode(...)` in Ethereum.
```
[the remark this answers is not in the document]
Comment (evan-forbes) [reply rpc/core/blocks.go:181]: yeah, agree

File: rpc/core/blocks.go [hunk 2]
```diff
@@ -177,59 +177,31 @@ func DataRootInclusionProof(
 	return &ctypes.ResultDataRootInclusionProof{Proof: *proof}, nil
 }
 
+// Pad bytes to given length
+func padBytes(byt []byte, length int) []byte {
+	l := len(byt)
+	if l == length {
```
[the remark this answers is not in the document]
Comment (amityadav0) [reply rpc/core/blocks.go:183]: yes, updating it.

File: rpc/core/blocks.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (amityadav0) [reply rpc/core/blocks.go:181]: got it.

File: rpc/core/blocks.go [hunk 3]
```diff
@@ -177,59 +177,31 @@ func DataRootInclusionProof(
...
-	}
-	packed, err := args.Pack(&tuple)
-	if err != nil {
-		return nil, err
+	hexRepresentation := strconv.FormatUint(height, 16)
+	// Make sure hex representation has even length
+	if len(hexRepresentation)%2 == 1 {
+		hexRepresentation = "0" + hexRepresentation
 	}
+	hexBytes, _ := hex.DecodeString(hexRepresentation)
```
[the remark this answers is not in the document]
Comment (amityadav0) [reply rpc/core/blocks.go]: done

**Comment by amityadav0** [comment]
@SweeXordious Thanks for the review. I have updated the code

File: rpc/core/blocks.go [hunk 4]
```diff
@@ -177,6 +177,24 @@ func DataRootInclusionProof(
 	return &ctypes.ResultDataRootInclusionProof{Proof: *proof}, nil
 }
 
+// Pad bytes to given length
```
[the remark this answers is not in the document]
Comment (amityadav0) [reply rpc/core/blocks.go:180]: Thanks

**Review by SweeXordious — APPROVED** [review]

**Pull request merged**